warehouse/tuf/services.py

```diff
@@ -0,0 +1,378 @@
+        Gets the Key based on role name and key type
+
+        Args:
+            rolename: role name
+            key_type: 'private' or 'public'
+
+        """
+        if key_type == "private":
+            privkey_path = os.path.join(self._key_path, f"{rolename}*")
+            role_keys = glob.glob(privkey_path)
+            keys_sslib = [
+                import_ed25519_privatekey_from_file(
+                    key, self._request.registry.settings[f"tuf.{rolename}.secret"]
+                )
+                for key in role_keys
+                if "pub" not in key
+            ]
+        elif key_type == "public":
+            pubkey_path = os.path.join(self._key_path, f"{rolename}*.pub")
+            role_keys = glob.glob(pubkey_path)
+            keys_sslib = [import_ed25519_publickey_from_file(key) for key in role_keys]
+        else:
+            raise ValueError(f"invalid key_type '{key_type}'")
+
+        return keys_sslib
+
+
+@implementer(IStorageService)
+class LocalStorageService:
+    def __init__(self, repo_path):
+        self._repo_path = repo_path
+
+    @classmethod
+    def create_service(cls, context, request):
+        return cls(
+            request.registry.settings["tuf.repo.path"],
+        )
+
+    @contextmanager
+    def get(self, role, version=None):
+        if role == Role.TIMESTAMP.value:
+            filename = os.path.join(self._repo_path, f"{role}.json")
+        else:
+            if version is None:
+                filenames = glob.glob(os.path.join(self._repo_path, f"*.{role}.json"))
+                versions = [
+                    int(name.split("/")[-1].split(".", 1)[0]) for name in filenames
+                ]
+                try:
+                    version = max(versions)
+                except ValueError:
+                    version = 1
+
+            filename = os.path.join(self._repo_path, f"{version}.{role}.json")
+            if not os.path.isfile(filename):
+                filename = os.path.join(self._repo_path, f"{role}.json")
+
+        file_object = None
+        try:
+            file_object = open(filename, "rb")
+            yield file_object
+        except OSError:
+            raise StorageError(f"Can't open {filename}")
+        finally:
+            if file_object is not None:
+                file_object.close()
+
+    def put(self, file_object, filename):
+        file_path = os.path.join(self._repo_path, filename)
+        if not file_object.closed:
+            file_object.seek(0)
+
+        try:
+            with open(file_path, "wb") as destination_file:
+                shutil.copyfileobj(file_object, destination_file)
+                destination_file.flush()
+                os.fsync(destination_file.fileno())
+        except OSError:
+            raise StorageError(f"Can't write file {filename}")
+
+    def store(self, file_object, filename):
+        self.put(file_object, filename)
+
+
+@implementer(IRepositoryService)
+class LocalRepositoryService:
```

Comment:
IIUC, this service could be instantiated with non-local `IStorageService` and `IKeyService` implementations too, right? Should we maybe call it just `RepositoryService`?

Reply:
Oh, I missed this comment. Sorry. 

Yes, the services are instantiated based on the Warehouse configuration.
I completely agree. I will change it. I don't see another "type"  of Repository.